Adding MaterialFormattedTextField

Pliavi opened the pull request.

Pliavi committed:
Adding MaterialFormattedTextField

leMaik commented:
Thanks for the PR! Couldn't this be done by inheriting from `MaterialTextField`?

Pliavi commented:
Maybe, the class made to the project was a fast adaptation. I'll try to see what makes the JFormattedTextField formattable (possibly an interface or abstract class) and add only the parts of it to use this inheritance.

leMaik commented:
@Pliavi That would be awesome!

Commit:
Parcial change

Pliavi committed:
Update to follow the other text components.

Pliavi committed:
Fixing NullPointerException on JFormattedTextField constructor.

Pliavi commented on src/main/java/de/craften/ui/swingmaterial/MaterialFormattedTextField.java:
```diff
@@ -63,8 +63,8 @@ public MaterialFormattedTextField(AbstractFormatter formatter) {
-		super(factory);
-		initMaterialFormattedTextField();
+		this();
+		setFormatterFactory(factory);
```
The setFormatterFactory()>setValue()>setFormatter() method uses setText() method, that call the FloatingLabel and Line update, but setFormatterFactory() was in the constructor, causing a nullPointerException.

Pliavi committed:
Fixing NullPointerException on JFormattedTextField constructor.
The setFormatterFactory()>setValue()>setFormatter() method uses setText() method, that call the FloatingLabel and Line update, but setFormatterFactory() was in the constructor, causing a nullPointerException.

leMaik commented:
Nice refactoring! :+1:
Please link the material specs (TextField) in the javadoc and feel free to mention yourself as `@author`. Is this ready to be merged?

Pliavi commented:
OK, i'll add when coming home.
It's not ready yet, this morning I've tested with default value constructor (the JFormattedTextField creates the MaskFormatter) and got another exception, possibly all the constructor will have to be rewrote like those with MaskFormatter in the signature.
But it can be corrected with 2 or 3 lines, today or tomorrow morning it'll be ready to merge.

Pliavi committed:
Fixing constructors
Fixed all the constructors and tested, now it's ready to be merged.

Pliavi commented:
Changes made, can be merged now
